packages/vitest/src/runtime/runners/test.ts

```diff
@@ -62,6 +62,14 @@ export class VitestTestRunner implements VitestRunner {
       }
 
       const result = await this.snapshotClient.finish(suite.file.filepath)
+      if (
+        this.workerState.config.snapshotOptions.updateSnapshot === 'none'
+        && result.unchecked
+      ) {
+        const error = new Error(`Obsolete snapshots found: ${result.uncheckedKeys.join(', ')}`)
+        suite.result!.errors ??= []
+        suite.result!.errors.push(error)
```

Comment:
The error needs to be serialized first. For example, 

https://github.com/vitest-dev/vitest/blob/ce2a06b7bde0e8acd25e262a97624c3c4d6a472e/packages/runner/src/utils/collect.ts#L103